include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -232,9 +232,18 @@ destroy(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __
     if constexpr (!::std::is_trivially_destructible_v<_ValueType>)
     {
 #if _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDWOS_BROKEN
-        oneapi::dpl::__internal::__pattern_walk1(::std::forward<_ExecutionPolicy>(__exec), __first, __last,
-                                                 [](_ReferenceType __val) { __val.~_ValueType(); },
-                                                 /*__is_vector*/ ::std::false_type{}, __is_parallel);
+        if constexpr (oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy>::value)
```

Comment:
Do you really need duplicate if/else branches?
May be better to eval one parameter value?

Reply:
Yes, I think this should be able to be collapsed back into a single line of code for the `__pattern_walk1` call, with another variable assigned based on `_ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN`.

Reply:
May be something like this?

```C++
template <class _ExecutionPolicy>
constexpr bool
eval_expr_for_destroy(_ExecutionPolicy __exec)
{
#if _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDWOS_BROKEN
    if constexpr (oneapi::dpl::__internal::__is_host_execution_policy<_ExecutionPolicy>::value)
        return ::std::false_type{};
    else
#endif
        return oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec);
}

// [specialized.destroy]

template <class _ExecutionPolicy, class _ForwardIterator>
oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy>
destroy(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last)
{
    typedef typename ::std::iterator_traits<_ForwardIterator>::value_type _ValueType;
    typedef typename ::std::iterator_traits<_ForwardIterator>::reference _ReferenceType;

    const auto __is_parallel =
        oneapi::dpl::__internal::__is_parallelization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec);

    oneapi::dpl::__internal::__pattern_walk1(::std::forward<_ExecutionPolicy>(__exec), __first, __last,
                                            [](_ReferenceType __val) { __val.~_ValueType(); },
                                            eval_expr_for_destroy<_ExecutionPolicy>(__exec), __is_parallel);
}
```

Reply:
I am worried that it would not work. `eval_expr_for_destroy` is not evaluated at compile-time, because `__exec` run-time argument. `eval_expr_for_destroy` must be a `constexpr` to avoid compilation of a broken configuration.

Reply:
I have followed the suggestion to remove `#else` branch.

Reply:
I had to revert that change, because if `#else` branch is removed, then the default one with broken configuration is always compiled...

Reply:
Thats a good point.  It needs to be remove from the code path as a constexpr to resolve the crash...

Reply:
And without `constexpr` is this variant useful?

Reply:
What about this:

```
    const auto __is_vector =
#if _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN
        ::std::conditional_t<
            oneapi::dpl::__internal::__is_host_execution_policy<::std::decay_t<_ExecutionPolicy>>::value,
            ::std::false_type,
            oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec)
        >;
#else
        oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec);
#endif // _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN

        oneapi::dpl::__internal::__pattern_walk1(::std::forward<_ExecutionPolicy>(__exec), __first, __last,
                                                 [](_ReferenceType __val) { __val.~_ValueType(); },
                                                 __is_vector, __is_parallel);
```

Reply:
Thanks for the idea, Alexey. I have changed the code accordingly with some minor corrections.